### PR TITLE
(WIP) per_page filter (#6)

### LINKED ISSUE #6: Filter perPage
Currently the `per_page` option is included in formation

Lets make it a filter to allow setting a component to the frontend.

```php
Filter::make('per_page')->per(),
```

Which uses a default config value when no parameters:

```
config/headless-laravel.php filters.per = [10, 25, 100]
```

```php
Filter::make('per_page')->per([10, 25, 100]),
```

```php
public function test_per_page()
{
    Post::factory(4)->create();

    $this->get('/posts?per_page=2')
        ->assertJsonCount(2, 'data')
        ->assertJsonPath('last_page', 2)
        ->assertJsonPath('total', 4);
}
```

### PR DESCRIPTION
**Pull request by mitulgolakiya**
closes #6

### CODE REVIEW

File: src/Traits/Findable.php [hunk 1]
```diff
@@ -2,9 +2,16 @@
...
+
 trait Findable
 {
     use Searchable;
     use Sortable;
     use Filterable;
+
+    public function getPerPage()
+    {
+        return Request::input('per_page', 15);
```
Comment (dillingham) [inline src/Traits/Findable.php:15]: This trait is only to add all the other traits

**Review by dillingham — CHANGES_REQUESTED** [review]

File: src/Traits/Findable.php (hunk 1 above)
Comment (mitulgolakiya) [reply src/Traits/Findable.php]: Removed.

File: src/Filter.php [hunk 2]
```diff
@@ -872,6 +879,13 @@ public function apply($query)
             throw new UnauthorizedException();
         }
 
+        if (!empty($this->per)) {
+            $query->getModel()->setPerPage($this->value);
```
Comment (dillingham) [inline src/Filter.php:883]: This is not how other filters work. 

Use withQuery and remove the class property $per

File: src/Filter.php [hunk 3]
```diff
@@ -1096,4 +1110,24 @@ public function meta(): array
             'modifiers' => $this->modifiers,
         ];
     }
+
+    public function per($per = []): self
```
Comment (dillingham) [inline src/Filter.php:1114]: Add array type hint

File: tests/FilterTest.php [hunk 4]
```diff
@@ -656,4 +656,14 @@ public function test_converting_to_cents()
             ->assertJsonCount(1, 'data')
             ->assertJsonPath('data.0.id', $two->id);
     }
+
+    public function test_per_page()
```
Comment (dillingham) [inline tests/FilterTest.php:660]: Add test for higher than max throwing a validation error

**Review by dillingham — CHANGES_REQUESTED** [review]

**Pull request closed without merging**